test/integration/console_sock_test.lua

```diff
@@ -20,11 +20,21 @@ g.before_all = function()
         http_port = 8080,
         cluster_cookie = 'super-cluster-cookie',
         env = {
-            TARANTOOL_CONSOLE_SOCK = fio.pathjoin(tmpdir, '/foo.sock')
+            TARANTOOL_CONSOLE_SOCK = fio.pathjoin(tmpdir, '/foo.sock'),
+            NOTIFY_SOCKET = fio.pathjoin(tmpdir, '/notify.sock')
         },
     })
+    local notify_socket = assert(socket('AF_UNIX', 'SOCK_DGRAM', 0), 'Can not create socket')
+    assert(notify_socket:bind('unix/', server.env.NOTIFY_SOCKET), notify_socket:error())
     server:start()
-    t.helpers.retrying({}, function() server:graphql({query = '{}'}) end)
+    t.helpers.retrying({}, function() 
+        while not notify_socket:readable(1) do
```

Comment:
This condition is wrong. You try to call recv when socket isn't readable. Don't implement a loop at all, `retrying` already implies it.